eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -886,14 +886,14 @@ object Helpers {
     def timedoutHtlcs(remoteCommit: RemoteCommit, remoteDustLimit: Satoshi, tx: Transaction)(implicit log: LoggingAdapter): Set[UpdateAddHtlc] =
       if (tx.txid == remoteCommit.txid) {
         // the tx is a commitment tx, we can immediately fail all dust htlcs (they don't have an output in the tx)
-        (remoteCommit.spec.htlcs.filter(_.direction == IN) -- Transactions.trimReceivedHtlcs(remoteDustLimit, remoteCommit.spec)).map(_.add)
+        (remoteCommit.spec.htlcs.collect(incoming) -- Transactions.trimReceivedHtlcs(remoteDustLimit, remoteCommit.spec).map(_.add))
       } else {
         // maybe this is a timeout tx, in that case we can resolve and fail the corresponding htlc
         tx.txIn.map(_.witness match {
           case ScriptWitness(Seq(remoteSig, ByteVector.empty, htlcReceivedScript)) =>
             val paymentHash160 = htlcReceivedScript.slice(69, 69 + 20)
             log.info(s"extracted paymentHash160=$paymentHash160 from tx=$tx (claim-htlc-timeout)")
-            remoteCommit.spec.htlcs.filter(_.direction == IN).map(_.add).filter(add => ripemd160(add.paymentHash) == paymentHash160)
+            remoteCommit.spec.htlcs.collect { case IncomingHtlc(add) if ripemd160(add.paymentHash) == paymentHash160 => add }
```

Comment:
nit: this could also use `.collect(incoming)` as above

Reply:
Done with cc438497f.